pandas/core/reshape/reshape.py

```diff
@@ -461,7 +461,17 @@ def factorize(index):
                                names=[frame.index.name, frame.columns.name],
                                verify_integrity=False)
 
-    new_values = frame.values.ravel()
+    # For homogonoues EAs, self.values will coerce to object. So
+    # we concatenate instead.
+    dtypes = list(frame.dtypes.values)
+    dtype = dtypes[0]
+    if frame._is_homogeneous_type and is_extension_array_dtype(dtype):
```

Comment:
you can do the is_extension_array_dtype(..) check after the frame._is_homogenouse_type (as you don't have to create the list of dtypes). its not a performance thing, rather you don't have to worry about this, unless you are non homogenous.